### PR TITLE
Bugfix: Avoid crash when "c" is pressed in Empty Narrow

### LINKED ISSUE #1453: ZT crashes when trying to send a new message to an empty stream upon pressing "c."
The problem arises in both the stable release and the development environment. The only modification I made was commenting out the line `self.focus_msg = focus_msg` to prevent the "Focus out of range" IndexError. I'm don't think the commented-out line has anything to do with error, but a confirmation would be nice. 

The commented out line is a common fix that was recommended in https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/IndexError.20while.20starting.2E.

**To reproduce the issue:**

1. Navigate to the right column view.
2. Select a user by pressing "Enter," which takes you to the compose box.
3. Press "Esc" to exit the compose box.
4. Press "c."

This sequence of actions crashes ZT.

**Traceback**
```
{e}


None
Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.10/site-packages/zulipterminal/cli/run.py", line 509, in main
    ).main()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/zulipterminal/core.py", line 678, in main
    self.loop.run()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/urwid/main_loop.py", line 287, in run
    self._run()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/urwid/main_loop.py", line 385, in _run
    self.event_loop.run()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/urwid/main_loop.py", line 790, in run
    self._loop()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/urwid/main_loop.py", line 827, in _loop
    self._watch_files[fd]()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/urwid/raw_display.py", line 416, in <lambda>
    wrapper = lambda: self.parse_input(
  File "/home/ubuntu/.local/lib/python3.10/site-packages/urwid/raw_display.py", line 515, in parse_input
    callback(processed, processed_codes)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/urwid/main_loop.py", line 412, in _update
    self.process_input(keys)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/urwid/main_loop.py", line 513, in process_input
    k = self._topmost_widget.keypress(self.screen_size, k)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/zulipterminal/ui.py", line 251, in keypress
    self.middle_column.keypress(size, key)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/zulipterminal/ui_tools/views.py", line 616, in keypress
    self.footer.stream_box_view(caption=stream_dict[stream_id]["name"])
KeyError: None
```

### PR DESCRIPTION
**Pull request by rsashank**
### What does this PR do, and why?
Fixes #1453 

Discussion:
https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/ZT.20crashes.20when.20pressing.20.22c.22.20in.20empty.20stream

### External discussion & connections

- [x] Discussed in **#zulip-terminal** in `ZT crashes when pressing "c" in empty stream`
- [x] Fully fixes #1453 
- [ ] Partially fixes issue #
- [ ] Builds upon previous unmerged work in PR #
- [ ] Is a follow-up to work in PR #
- [ ] Requires merge of PR #
- [ ] Merge will enable work on #

### How did you test this?

- [x] Manually - Behavioral changes
- [x] Manually - Visual changes
- [x] Adapting existing automated tests
- [ ] Adding automated tests for new behavior (or missing tests)
- [x] Existing automated tests should already cover this (*only a refactor of tested code*)

### Self-review checklist for each commit
- [x] It is a [minimal coherent idea](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development)
- [ ] It has a commit summary following the [documented style](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development) (title & body)
- [ ] It has a commit summary describing the  motivation and reasoning for the change
- [x] It individually passes linting and tests
- [] It contains test additions for any new behavior
- [x] It flows clearly from a previous branch commit, and/or prepares for the next commit

### CODE REVIEW

**Comment by rsashank** [comment]
@zulipbot add "PR needs review"

File: zulipterminal/ui_tools/views.py [hunk 1]
```diff
@@ -591,7 +591,11 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             if self.footer.focus is None:
                 stream_id = self.model.stream_id
                 stream_dict = self.model.stream_dict
-                self.footer.stream_box_view(caption=stream_dict[stream_id]["name"])
+                if stream_id is None:
+                    self.model.controller.report_error(["This is not a valid stream."])
+                    return key
+                else:
+                    self.footer.stream_box_view(caption=stream_dict[stream_id]["name"])
```
[the remark this answers is not in the document]
Comment (rsashank) [reply zulipterminal/ui_tools/views.py:597]: I've put the call without passing a caption. 

I'm not too sure as to why this is only an issue for empty narrows as of now, but I'll look into it and keep you updated on #zulip-terminal.

**Comment by rsashank** [comment]
@zulipbot add "PR needs review"
@zulipbot remove "PR awaiting update"

**Comment by neiljp** [comment]
@rsashank Thanks for this updated bugfix - this avoids the crash and performs as expected :+1: I just merged it after a slight update to the commit text :tada:

As noted in the stream, this would benefit from a cleaner fuller resolution in terms of the placeholder for the empty message list, though that is a bigger change to integrate!

**Pull request merged**